File: csharp/darts/Darts.cs
using System;

public static class Darts
{
  public static int Score(double x, double y)
  {
    var furthest = Math.Max(x, y);

    if (furthest > 10) return 0;
    if (furthest > 5) return 1;
    if (furthest > 1) return 5;

    return 10;
  }
}

File: ruby/darts/darts.rb
class Darts
  def initialize(x, y)
    @distance = Math.hypot(x, y)
  end

  def score
    if @distance <= 1
      10
    elsif @distance <= 5
      5
    elsif @distance <= 10
      1
    else
      0
    end
  end
end
